fractions/microprofile/microprofile-restclient/README.adoc

```diff
@@ -0,0 +1,42 @@
+= MicroProfile Rest Client
+
+== CDI Interceptors Support
+
+In general, Rest Client proxies are not created by the CDI container and therefore method invocations do not pass through CDI interceptors.
+In Thorntail, however, you can associate business method interceptors (denoted by `@AroundInvoke` annotation) with a Rest Client proxy.
+You can do so by using interceptor bindings.
+This feature is non-portable.
+The primary use case is the support of xref:microprofile-fault-tolerance[] annotations, for example:
+
+[source,java]
+----
+import org.eclipse.microprofile.faulttolerance.Retry;
+
+@Path("/v1")
+interface MyClient {
+
+    @Retry(maxRetries = 3) // Retry on any exception thrown
+    @GET
+    @Path("/hello")
+    String hello();
+}
+----
+
+NOTE: The `org.eclipse.microprofile.faulttolerance.Asynchronous` is currently not supported because the underlying RESTEasy client is not able to handle the `java.util.concurrent.Future` return types.
+
+== RestClientProxy
+
+Additionaly, every Rest Client proxy implements `org.wildfly.swarm.microprofile.restclient.RestClientProxy` interface which allows to:
```

Comment:
Please clarify in addition to what does the proxy implement the interface or remove the word _Additionally._

Please add _you_ after _allows._

Reply:
> Please clarify in addition...

"Normal" Rest Client proxies are not required to implement any other interface than a concrete Rest Client interface (MyClient in the example above). I think this is obvious for any MP Rest Client user.

> Please add you after allows.

Uf, why? Sure I can add `you` there but it's not necessary and does not add any value.

Reply:
>> Please clarify in addition...
>
> "Normal" Rest Client proxies are not required to implement any other interface than a concrete Rest Client interface (MyClient in the example above). I think this is obvious for any MP Rest Client user.

I don't think that's obvious. I'd be in favor of something like "In addition to MP Rest Client specification, ...".

Reply:
> "Normal" Rest Client proxies are not required to implement any other interface than a concrete Rest Client interface (MyClient in the example above).

I get that this text is meant for someone with knowledge of MP, but I have fundamental problems understanding the sentence. In the doc, you say that _every Rest Client proxy implements the ... interface,_ but your explanation in the comment above implies that normally, they don't. Does it mean that only the Thorntail implementation implements that functionality, or something else?

Reply:
> Uf, why?

It's not a grammatically correct sentence. _Allow_ is a transitive verb, so it requires an object _(you)._ Alternatively, it could be followed by a gerund _(it allows obtaining... and releasing...)._

Reply:
@tradej [Got it](https://english.stackexchange.com/questions/85069/is-the-construction-it-allows-to-proper-english). Thanks for explanation ;-).